server/src/main/scala/actors/DrtSystem.scala

```diff
@@ -333,48 +322,34 @@ case class DrtSystem(actorSystem: ActorSystem, config: Configuration, airportCon
       arrivalsBaseSource = baseArrivalsSource(),
       arrivalsFcstSource = forecastArrivalsSource(airportConfig.feedPortCode),
       arrivalsLiveSource = liveArrivalsSource(airportConfig.feedPortCode),
+      initialShifts = initialState(shiftsActor, GetState).getOrElse(ShiftAssignments(Seq())),
+      initialFixedPoints = initialState(fixedPointsActor, GetState).getOrElse(FixedPointAssignments(Seq())),
+      initialStaffMovements = initialState[StaffMovements](staffMovementsActor, GetState).map(_.movements).getOrElse(Seq[StaffMovement]()),
       recrunchOnStart = recrunchOnStart
     ))
     crunchInputs
   }
 
-  def initialRegisteredArrivals(askableRegisteredArrivalsActor: AskableActorRef): Future[Option[RegisteredArrivals]] = {
-    askableRegisteredArrivalsActor.ask(GetState)(new Timeout(5 minutes)).map {
-      case ra: RegisteredArrivals =>
-        system.log.info(s"Got initial registered arrival state from ${askableRegisteredArrivalsActor.toString}")
-        Option(ra)
-      case _ =>
-        system.log.warning(s"Got no initial registered arrival state from ${askableRegisteredArrivalsActor.toString}")
+  def initialState[A](askableActor: AskableActorRef, toAsk: Any): Option[A] = Await.result(initialStateFuture[A](askableActor, toAsk), 5 minutes)
+
+  def initialStateFuture[A](askableActor: AskableActorRef, toAsk: Any): Future[Option[A]] = {
+    val future = askableActor.ask(toAsk)(new Timeout(5 minutes)).map {
+      case Some(state: A) if state.isInstanceOf[A] =>
+        log.info(s"Got initial state (Some(${state.getClass})) from ${askableActor.toString}")
+        Option(state)
+      case state: A if !state.isInstanceOf[Option[A]] =>
```

Comment:
I think the `if` part of this condition must always be true.
`case state: A` will only be matched if state is an instance of `A` so therefore cannot be an instance of Option[A]

unless I'm missing something here...

Reply:
It doesn't work without  the `if`. I presume it's down to type erasure but adding the if was the only way I could get it to correctly match

Reply:
you can probably get away with 
```
case state if !state.isInstanceOf[Option[A]] =>
```
which would be less confusing maybe?

Reply:
(you'd probably also want to put this one last)

Reply:
you actually can't because it then complains that the return type is not correct